core/templates/components/button-directives/hint-and-solution-buttons.directive.html

```diff
@@ -26,22 +23,115 @@
       </button>
     </li>
   </ul>
+  <div ng-class="{'remove-hint-box': !$ctrl.isVisible}">
+    <div class="hint-box"
+         ng-if="$ctrl.isVisible && $ctrl.isTooltipVisible()">
+      Need help? View a hint for this problem!
+      <button type="button"
+              class="btn close"
+              aria-label="Close"
+              ng-click="$ctrl.isVisible = !$ctrl.isVisible">
+        <span aria-hidden="true">&times;</span>
+      </button>
+    </div>
+  </div>
 </div>
 
 <style>
+  .close {
+    bottom: 10px;
+    font-size: 22px;
+    left: 115px;
+    margin-left: 10px;
+    position: relative;
+  }
+
+  .hint-box {
+    -webkit-animation: bounce 4s infinite;
+    animation: bounce 4s infinite;
+    background-color: #f0ffff;
+    border: 1px solid #2d4a9d;
+    border-radius: 5px;
+    bottom: 65px;
+    box-shadow: 0px 0px 6px #2d4a9d;
+    color: #2d4a9d;
+    font-size: 17px;
+    height: 40px;
+    left: -3px;
+    padding: 20px 0px 20px 10px;
+    position: relative;
+    text-align: center;
+    white-space: nowrap;
+    width: 395px;
+    z-index: 2;
+  }
+
+  .remove-hint-box {
+    display: none;
+  }
+
+  .hint-box::after,
+  .hint-box::before {
+    border: solid transparent;
+    content: " ";
+    left: 50%;
+    position: absolute;
+    top: 100%;
+  }
+
+  .hint-box::after {
+    border-top: 23px solid #f0ffff !important;
```

Comment:
Here and below, please avoid `!important` (see [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide#css)).

Reply:
For the desired styling, the `border-width` should be before the `boder-top` but to pass the lint check I had to reorder them. If you can suggest some alternative, would be helpful.

Reply:
Hm, I'm slightly confused. Aren't both CSS rules (`border-width`, `border-top`) independent of each other, why would they create a conflict?

Reply:
Done.